ob-pattern-1/templates/wso2ob-pattern-1-obam-ingress.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wso2ob-pattern-1-obam-ingress
+  namespace: {{ .Release.namespace }}
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    nginx.ingress.kubernetes.io/ssl-passthrough: "true"
+    nginx.ingress.kubernetes.io/affinity: "cookie"
+    nginx.ingress.kubernetes.io/session-cookie-name: "route"
+    nginx.ingress.kubernetes.io/session-cookie-hash: "sha1"
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+spec:
+  tls:
+  - hosts:
+    - {{ .Values.wso2.deployment.wso2obam.hostname  }}
+    - {{ .Values.wso2.deployment.wso2obam.gateway.hostname  }}
```

Comment:
I'll create a new ingress for gateway along with above annotations. Shouldn't we use 
```nginx.ingress.kubernetes.io/backend-protocol```
to make sure nginx communicate with the backend with HTTPS ?

Reply:
The annotations I have provided are applicable for the Kubernetes NGINX Ingress Controller `nginx-0.22.0` release.

@kdanW it would be great if you can test this with the latest release (as of now `nginx-0.26.1`) and check. Yes, I believe the annotation you had pointed out is required for releases after `nginx-0.22.0`.

Reply:
@chirangaalwis shall i finish doing the complete setup and testing with ```nginx-0.22.0``` and then work on updating to latest version?

Reply:
+1

Reply:
Added a separate ingress controller for the gateway under https://github.com/wso2/kubernetes-open-banking/pull/13/commits/703ce07315a741da7d4ed51b68200588d8d92c10